### PR TITLE
use native ProxyGet from client-go

### PR DESCRIPTION
**Pull request by shamsher31**
As `ProxyGet` is now natively available in client-go https://github.com/kubernetes/kubernetes/pull/92251

Part of https://github.com/istio/istio/issues/25119

### CODE REVIEW

**Comment by shamsher31** [comment]
Agree, that's why I put it in the draft, will update it once the next stable release is available.

**Comment by howardjohn** [comment]
Not stale - 1.19 comes out soon

**Comment by shamsher31** [comment]
Holding until `pkg/log/null.go:48:32: undefined: logr.InfoLogger` issue is fixed in `controller-runtime`. 

Issue: https://github.com/kubernetes-sigs/controller-runtime/issues/1033, Fixed: https://github.com/kubernetes-sigs/controller-runtime/pull/1011

Once it is available in `controller-runtime` 1.19 https://github.com/kubernetes-sigs/controller-runtime/pull/1068, I can test it.

**Comment by howardjohn** [comment]
This should be ready to go with a rebase

File: pkg/kube/client.go [hunk 1]
```diff
@@ -451,28 +449,6 @@ func (c *client) PodLogs(ctx context.Context, podName, podNamespace, container s
 	return builder.String(), nil
 }
```
Comment (ostromart) [inline pkg/kube/client.go:451]: could we retain this function (as a wrapper) and make it public? i need the function for bug-report tool and it's probably generally useful.

**Comment by shamsher31** [comment]
> Looks good but seems like the tests that use this are failing.

`proxy-status` does not work after upgrading. Maybe it requires `params`. I'm looking into it.
```
$ istioctl proxy-status details-v1-558b8b4b76-8bhkf.default
Problem forwarding to istiod-6f5fd7cb8f-w9vpw.istio-system: the server is currently unable to handle the request (get pods https:istiod-6f5fd7cb8f-w9vpw:15014); skipping.Error: 1 error occurred:
        * the server is currently unable to handle the request (get pods https:istiod-6f5fd7cb8f-w9vpw:15014)
```

**Comment by shamsher31** [comment]
> ```
> $ istioctl proxy-status details-v1-558b8b4b76-8bhkf.default
> Problem forwarding to istiod-6f5fd7cb8f-w9vpw.istio-system: the server could not find the requested resource (get pods istiod-6f5fd7cb8f-w9vpw:15014); skipping.Error: 1 error occurred:
>         * the server could not find the requested resource (get pods istiod-6f5fd7cb8f-w9vpw:15014)
> ```

@howardjohn  Is this because of `istio-proxy` container is missing from `istiod` pod, so the `ProxyGet` is not able to reach the pod?


`$ ./out/linux_amd64/istioctl version`  works fine [with same call](https://github.com/istio/istio/pull/25241/files#diff-a483d6c0863082c3df21f4aad513afe2R561) because other pods have `istio-proxy` container.

I remember I created an issue for the missing `istio-proxy` container from `istiod` but couldn't find it.

File: pkg/kube/client.go (hunk 1 above)
Comment (shamsher31) [reply pkg/kube/client.go:458]: @ostromart ProxyGet is now natively available so you can directly use it. eg:
```
c.CoreV1().Pods(pod.Namespace).ProxyGet("", pod.Name, "15014", "/version", nil).DoRaw(ctx)

c.CoreV1().Pods(istiod.Namespace).ProxyGet("", istiod.Name, "15014", path, nil).DoRaw(ctx)
```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**